### ISSUE TITLE
The message property of SlashContext object is never populated[BUG] 

### ISSUE THREAD

**Issue by Taiiwo** [interactions-py/interactions.py#356]
Trying to implement slash commands with my existing bot. Basically just looking to pass the message object through to an existing function. I set up the slash commands just fine, but when the callback executes, the message property of the context object is None. Looking at the code it looks like it's only populated after the message is sent.

My bot has a custom interface for registering commands, I'm not using the discord commands ext. I realise this code won't mean much out of context, but I figured providing it might help if I'm doing something wrong. Basically, an interface represents a single command. interface.func is the callback function that takes a formatted message object and performs some task, like sending a response. The interface has subcommands which are their own nested interfaces, and the flags and arguments are processed separately.

All I need is the message object of the invoking message, so I can pass it to my framework and process it as normal. Does it have something to do with the deferred status of the message? Is there a way to bypass it and access the message object before it's finished being deferred? Or just disable it alltogether?

Thanks, sorry for the kind of chaotic bug report. Please ask any questions you need

```python
# map message callback function to slash callback
def make_async(f):
    async def a(ctx, *args, **kwargs):
        # this runs when the slash command is executed
        ### ctx.message == None ###
        return f(self.format_message(ctx.message), *args, **kwargs)
    return a
# register the command
self.slash.add_slash_command(
    make_async(interface.func), # map the message callback to a slashcommand callback
    interface.name, interface.desc[:100], options=[create_option(
        "arguments", "The arguments and flags to supply to the command", option_type=1, required=False)]
)
# register the subcommands
for subcommand in interface.subcommands:
    self.slash.add_subcommand(
        make_async(subcommand.func),
        interface.name, name=subcommand.name, description=subcommand.desc[:100], options=[
            create_option(
                "arguments", "The arguments and flags to supply to the command",
                option_type=3, required=False
            )
        ]
    )```

**Comment by FayeDel** [comment]
Hiya!

So `Looking at the code it looks like it's only populated after the message is sent.` is intentional, but that's not because of this library, this is more of a Discord bit; when you do a slash command, there's no "message" to reference, compared to d.py's message saying, for example, `-say 1234` (If the slash equivalent, let's say, `/say 1234`, command and args is sent, there's no message symbolising the command itself, but as you said earlier, only the message that the code sends after with `ctx.send`).
In a lower level, I suppose, the message of the invocation command exists in d.py because it relies on message-based events (like Message intents), and slash commands are its own event (that doesn't utilise message responses)

Basically, the message object of the invoking message doesn't really *exist* (can't get `/say 1234` but you can get the message from `await ctx.send("1234")`

**Comment by ActuallyFlamey** [comment]
The SlashContext.message is the interaction RESPONSE, because a Slash Command invocation isn't a message.

Therefore, if you haven't sent anything yet, it will be None.

**Comment by Taiiwo** [comment]
Yea, that's what I figured. I guess I'm just going to have to reconstruct a message-looking object to pass back into my framework and just cross my fingers. Thanks for the fast response :)
